File: src/SPS.AdminWeb/src/types/notification.ts
// Notification related types based on backend DTOs

export interface NotificationResponse {
  id: number;
  type: number;
  category?: string;
  title?: string;
  content?: string;
  sendId?: string;
  recipient?: string;
  read: boolean;
  expiration?: string;
  createdTime: string;
  updatedTime: string;
}

export interface CreateNotificationRequest {
  type: number;
  category?: string;
  title: string;
  content: string;
  sendId?: string;
  recipient: string;
  expiration?: string;
}

export interface UnreadCountResponse {
  count: number;
}

File: src/SPS.Domain/Entities/Notification.cs
using SPS.Domain.Common;
using SPS.Domain.Enums;

namespace SPS.Domain.Entities;

public class Notification : BaseEntity<long>
{
    public NotificationType Type { get; set; }
    public string? Category { get; set; }
    public string? Title { get; set; }
    public string? Content { get; set; }
    public string? SendId { get; set; }
    public string? Recipient { get; set; }
    public bool Read { get; set; }
    public DateTime? Expiration { get; set; }
}
